index.bs

```diff
@@ -535,101 +539,233 @@ dictionary WriteParams {
+      {{WriteParams/type}}: {{WriteCommandType/"write"}},
+      {{WriteParams/position}}: |position|,
+      {{WriteParams/data}}: |data| })
+  :: Writes the content of |data| into the file associated with |stream| at |position|
+     bytes from the top of the file. Also updates the current file cursor offset to the
+     end of the written data.
+
+     No changes are written to the actual file on disk until the stream has been closed.
+     Changes are typically written to a temporary file instead.
+
+  : await |stream| . {{FileSystemWritableFileStream/write()|write}}({
+      {{WriteParams/type}}: {{WriteCommandType/"seek"}},
+      {{WriteParams/position}}: |position| })
+  :: Updates the current file cursor offset the |position| bytes from the top of the file.
+
+  : await |stream| . {{FileSystemWritableFileStream/write()|write}}({
+      {{WriteParams/type}}: {{WriteCommandType/"truncate"}},
+      {{WriteParams/size}}: |size| })
   :: Resizes the file associated with |stream| to be |size| bytes long. If |size| is larger than
      the current file size this pads the file with null bytes, otherwise it truncates the file.
 
      The file cursor is updated when {{truncate}} is called. If the offset is smaller than offset,
      it remains unchanged. If the offset is larger than |size|, the offset is set to |size| to
      ensure that subsequent writes do not error.
 
-     No changes are written to the actual file until on disk until {{FileSystemWritableFileStream/close()}}
-     is called. Changes are typically written to a temporary file instead.
+     No changes are written to the actual file until on disk until the stream has been closed.
+     Changes are typically written to a temporary file instead.
 </div>
 
 <div algorithm>
-The <dfn method for=FileSystemWritableFileStream>truncate(|size|)</dfn> method, when invoked, must run these
+The <dfn method for=FileSystemWritableFileStream>write(|data|)</dfn> method, when invoked, must run
+these steps:
+
+1. Let |writer| be the result of calling [$AcquireWritableStreamDefaultWriter$](<b>[=this=]</b>).
+1. Let |result| be [$WritableStreamDefaultWriterWrite$](|writer|, |data|).
+1. Perform [$WritableStreamDefaultWriterRelease$](|writer|).
+1. Return |result|.
+
+</div>
+
+### The {{FileSystemWritableFileStream/seek()}} method ### {#api-filesystemwritablefilestream-seek}
+
+<div class="note domintro">
+  : await |stream| . {{FileSystemWritableFileStream/seek()|seek}}(|position|)
+  :: Updates the current file cursor offset the |position| bytes from the top of the file.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemWritableFileStream>seek(|position|)</dfn> method, when invoked, must run these
 steps:
 
-1. TODO
+1. Let |writer| be the result of calling [$AcquireWritableStreamDefaultWriter$](<b>[=this=]</b>).
+1. Let |result| be [$WritableStreamDefaultWriterWrite$](|writer|, <code>{type: {{WriteCommandType/"seek"}}, position: |position|}</code>).
```

Comment:
You should keep in mind that when this object is converted to a `WriteParams` dictionary later, it will access the global`Object.prototype.size` and `Object.prototype.data`. There may be getters on these properties that execute arbitrary JavaScript. 

It's probably worth including a non-normative note about this, as it may have security implications for the implementor.

Alternatively, you could explicitly set `size` and `data` to undefined here, to avoid the possible side-effects.